src/license.rs

```diff
@@ -46,8 +53,16 @@ pub fn file(crate_root: &Path, rel_dir: &Path, license_name: &str, single_licens
     let lic_abs_path = crate_root.join(lic_path);
     let spec_abs_path = crate_root.join(spec_path);
     let simple_abs_path = crate_root.join(simple_path);
+    let custom_abs_path = crate_root.join(license_path.unwrap_or(Path::new("")));
 
-    if lic_abs_path.exists() {
+    if license_path.is_some() && custom_abs_path.exists() {
+        let md5sum = file_md5(custom_abs_path).unwrap_or_else(|_| String::from("generateme"));
+        format!(
+            "file://{};md5={} \\\n",
+            rel_dir.join(license_path.unwrap()).parse_dot().unwrap().display(),
```

Comment:
instead of unwrapping `license_path` here you should access it by extracting it in the `if` as I suggested above already for the assignment.

is license_path user supplied? if yes then you shouldn't unwrap `parse_dot()` result.

Reply:
> instead of unwrapping license_path here you should access it by extracting it in the if as I suggested above already for the assignment.

I tried and got:
```
error[E0658]: `let` expressions in this position are unstable
  --> src/license.rs:58:8
   |
58 |     if let Some(lp) = license_path && custom_abs_path.exists() {
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
```

> is license_path user supplied?

Yes. However, I do not know what to do if `unwrap()` fails, it shouldn't. What do you suggest?

Reply:
> I tried and got:

Yes, you cannot have both in the same `if` condition. I'm sure you know how to refactor that ;)

> Yes. However, I do not know what to do if parse_dot() fails, it shouldn't. What do you suggest?

If you can guarantee that it doesn't fail then you should use `unwrap()` otherwise `?`.